otherlibs/stdune/src/ansi_color.ml

```diff
@@ -316,20 +386,61 @@ let parse_line str styles =
                  the same as "\027[0m" by terminals *)
               []
             else
+              let style_of_string accu s =
```

Comment:
This `else` clause is not a good place to allocate all of these closures. They should be hoisted to the top.